src/modules/scraper/adapter/messaging/BlocksEventsConsumer.ts

```diff
@@ -242,11 +244,18 @@ export class BlocksEventsConsumer {
     }
   }
 
-  private fromFundsDepositedEventToDeposit(event: Event) {
+  private async fromFundsDepositedEventToDeposit(chainId: number, event: Event) {
     const typedEvent = event as FundsDepositedEvent2 | FundsDepositedEvent2_5;
     const { transactionHash, blockNumber } = typedEvent;
     const { depositId, originChainId, destinationChainId, amount, originToken, depositor, relayerFeePct, recipient } =
       event.args;
+    // In some cases the depositor is the txn msg.sender
+    let trueDepositor = depositor;
+
+    if (depositor === SPOKE_POOL_VERIFIER_CONTRACT_ADDRESS) {
+      const tx = await this.providers.getProvider(chainId).getTransaction(transactionHash);
```

Comment:
I believe that we already store the receipt in the db 🤔 Could be worth to use `getCachedTransactionReceipt`?

Reply:
You are right! 👍